main.cpp

```diff
@@ -18,13 +18,34 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	const auto* CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	  if (CastExpr ->getCastKind() == CK_ToVoid)
+            return;
+	  if (CastExpr ->getExprLoc().isMacroID())
+            return;
+	  auto ReplaceRange = CharSourceRange::getCharRange(
+      		CastExpr->getLParenLoc(), CastExpr->getSubExprAsWritten()->getBeginLoc());

```

Comment:
Please describe the difference between `getSubExprAsWritten()->getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)`.

Reply:
> Using getSubExprAsWritten()->getBeginLoc() all spaces, tabs, and line breaks are located by the location of the first symbol of the expression, for example, initial version:

`int main() {
    float exp;
    int i = (int)                     exp ;
    return 0;
}`
With CastExpr->getSubExprAsWritten()->getBeginLoc():
`int main() {
    float exp;
    int i = static_cast<int>(exp) ;
    return 0;
}`

> Using CastExpr->getRParenLoc().getLocWithOffset(1) all symbols located between (<type>) and expression will be move inside brackets after static_cast<type>. For example, 
`int main() {
    float exp;
    int i = static_cast<int>(                     exp) ;
    return 0;
}`

The C-style cast replacement problem is solved in both ways, so getSubExprAsWritten() -> getBeginLoc() and getRParenLoc().getLocWithOffset(1) is equivalen for this example.